scripts/kconfig/kconfigfunctions.py

```diff
@@ -888,17 +888,51 @@ def dt_nodelabel_array_prop_has_val(kconf, _, label, prop, val):
     If the node exists, it checks if the node node has a property
     'prop' with type "array". If so, and the property contains
     an element equal to the integer 'val', it returns "y".
+    If the property is of type "string-array", it checks if 'val' is
+    one of the strings in the array, returning "y" if so.
     Otherwise, it returns "n".
     """
     if doc_mode or edt is None:
         return "n"
 
     node = edt.label2node.get(label)
 
-    if not node or (prop not in node.props) or (node.props[prop].type != "array"):
+    if not node or (prop not in node.props):
         return "n"
+    elif node.props[prop].type == "array":
+        return "y" if int(val, base=0) in node.props[prop].val else "n"
+    elif node.props[prop].type == "string-array":
+        return "y" if val in node.props[prop].val else "n"
     else:
+        return "n"
+
+
+def dt_node_array_prop_has_val(kconf, _, path, prop, val):
+    """
+    This function looks for a node at 'path'.
+    If the node exists, it checks if the node has a property
+    'prop' with type "array". If so, and the property contains
+    an element equal to the integer 'val', it returns "y".
+    If the property is of type "string-array", it checks if 'val' is
+    one of the strings in the array, returning "y" if so.
+    Otherwise, it returns "n".
+    """
+    if doc_mode or edt is None:
+        return "n"
+
+    try:
+        node = edt.get_node(path)
+    except edtlib.EDTError:
+        return "n"
+
+    if not node or (prop not in node.props):
```

Comment:
The function bodies introduced in this patch have a fair bit of duplicated code. Please factor out the common code into a single place.